exp/controllers/awsmachinepool_controller.go

```diff
@@ -298,6 +298,11 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 		return nil
 	}
 
+	if err := r.reconcileLifecycleHooks(machinePoolScope, asgsvc); err != nil {
```

Comment:
We have a `ctx` variable in this function, which we don't pass here, but later down the stack we end up creating a `context.TODO()`. May be worth passing the context that we already have. What do you think?

Reply:
@fiunchinho it looks like most of the other interfaces for the ASGInterface and EC2Interface use the same pattern (they get called from places that have context, and they themselves create their own context.TODO context).
We could start breaking the pattern here, it'd be a bit of a divergence to the rest of the code

Reply:
Then for consistency sake, it'd be better to follow the same pattern for now. It could be addressed in a different PR later on

Reply:
I think creating a `context.TODO` might have been a mistake when usage of `*WithContext` AWS SDK functions was introduced. Context should always be specified where possible in order to support timeouts, for instance. Some other interface functions are correctly taking such an argument already.